addons/website_event/tests/test_event_internals.py

```diff
@@ -115,6 +115,40 @@ def test_process_attendees_form(self):
                 (0, 0, {'question_id': self.event_question_3.id, 'value_text_box': 'Free Text'})]}
         ])
 
+    def test_process_attendees_form_no_tickets(self):
+        """Check that registering with no ticket works."""
+        event = self.env['event.event'].create({
+            'name': 'Event Update Type',
+            'event_type_id': self.event_type_questions.with_user(self.env.user).id,
+            'date_begin': FieldsDatetime.to_string(datetime.today() + timedelta(days=1)),
+            'date_end': FieldsDatetime.to_string(datetime.today() + timedelta(days=15)),
+        })
+
+        name_question, email_question, _ = event.question_ids.filtered(
+            lambda q: q.question_type in ('name', 'email', 'phone', 'company_name'))
+
+        form_details = {
+            '1-name-%s' % name_question.id: 'Test',
+            '1-email-%s' % email_question.id: 'test@test.lan',
```

Comment:
"Attendee Name" and "attendee@example.com" ?

It helps making the below assertion a bit more meaningful, especially for the name.